src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,84 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.ValidationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private static User userOne;
+    private static User userTwo;
+    private static User userThree;
+    private static User nullUser;
+    private static User userFour;
+
+    @BeforeAll
+    public static void createEntities() {
+        registrationService = new RegistrationServiceImpl();
+        userOne = new User();
+        userTwo = new User();
+        userThree = new User();
+        userFour = new User();
+        nullUser = null;
+
+        //create first User with valid credentials
+        userOne.setLogin("John");
+        userOne.setAge(18);
+        userOne.setPassword("qwerty");
+
+        //create second User with invalid password
+        userTwo.setLogin("Corey");
+        userTwo.setAge(42);
+        userTwo.setPassword("asdf");
+
+        //create third User with invalid age
+        userThree.setLogin("John Deere");
+        userThree.setAge(12);
+        userThree.setPassword("qwerty");
+
+        //create fourth User with valid credentials
+        userFour.setLogin("James");
+        userFour.setAge(58);
+        userFour.setPassword("metallica");
+    }
+
+    @AfterEach
+    void clearStorage() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void registerUsersWithInvalidCredentials_notOk() {
+        assertThrows(ValidationException.class, () -> registrationService.register(userTwo));
+        assertThrows(ValidationException.class, () -> registrationService.register(userThree));
+        assertThrows(ValidationException.class, () -> registrationService.register(nullUser));
```

Comment:
Also, don't create test methods, that test different scenarios. One test per scenario.

For example: if you cannot create a new user because of his age - make it in one test.